flexmeasures/ui/templates/crud/account.html

```diff
@@ -276,69 +276,100 @@ <h3>Assets</h3>
-                    class="{{ asset.generic_asset_type.name | asset_icon }} left-icon"
-                    >{{ asset.name }}</i
-                  >
-                </td>
-                <td>
-                  {% if asset.latitude and asset.longitude %} LAT: {{
-                  "{:,.4f}".format( asset.latitude ) }} LONG: {{
-                  "{:,.4f}".format( asset.longitude ) }} {% endif %}
-                </td>
-                <td>{{ asset.id }}</td>
-                <td>
-                  {% if asset.owner %} {{ asset.owner.name }} {% else %} PUBLIC
-                  {% endif %}
-                </td>
-                <td>{{ asset.sensors | length }}</td>
-                <td>
-                  <a href="/assets/{{ asset.id }}/status">
-                    <button type="button" class="btn">Status</button>
-                  </a>
-                </td>
-                <td class="text-right"></td>
-                <td class="d-none">/assets/{{ asset.id }}</td>
-              </tr>
-              {% endfor %}
-            </tbody>
-          </table>
+            id="assetTable"
+          ></table>
         </div>
       </div>
     </div>
     <div class="col-md-2"></div>
   </div>
 </div>
-{% block paginate_tables_script %} {{ super() }} {% endblock %}
+
+<script>
+  const asset_icon_map = JSON.parse("{{ asset_icon_map | tojson | safe }}");
+
+  function Asset(id, name, account, latitude, longitude, sensors, asset_type) {
+    let icon = asset_icon_map[asset_type.toLowerCase()];
+    if (icon === undefined) icon = `icon-${asset_type}`;
+
+    this.name = `
+    <i class="${icon} left-icon">${name}</i>
+  `;
+
+    this.id = id;
+    this.location = "";
+    this.url = `/assets/${id}`;
+    this.status = `
+        <a href="/assets/${id}/status">
+          <button type="button" class="btn btn-primary">Status</button>
+        </a>
+      `;
+
+    if (account == null) this.owner = "PUBLIC";
+    else
+      this.owner = `
+          <a href="/accounts/${account["id"]}" title="View this account">${account["name"]}</a>
+        `;
+
+    this.num_sensors = sensors.length;
+
+    if (latitude != null && longitude != null)
+      this.location = `LAT: ${latitude}, LONG: ${longitude}`;
+  }
+
+  $(document).ready(function () {
+    $("#assetTable").dataTable({
+      serverSide: true,
+      columns: [
+        { data: "id", title: "Asset ID" },
+        { data: "name", title: "Name" },
+        { data: "owner", title: "Account" },
+        { data: "location", title: "Location" },
+        { data: "num_sensors", title: "Sensors" },
+        { data: "status", title: "Status" },
+        { data: "url", title: "URL", className: "d-none" },
+      ],
+      ajax: function (data, callback, settings) {
+        let filter = data["search"]["value"];
+        let url = `{{url_for("AssetAPI:index")}}?page=${
+          Math.floor(data["start"] / data["length"]) + 1
+        }&per_page=${data["length"]}&all_accessible=true`;
```

Comment:
Why `all_accessible=true`? Here, we really only care about the assets in the account. So instead you should add the `account` parameter.